lib/tooling/llvm-mca-tool.ts

```diff
@@ -47,21 +47,64 @@ export class LLVMMcaTool extends BaseTool {
 
     override async runTool(compilationInfo: Record<any, any>, inputFilepath?: string, args?: string[]) {
         const isets = new InstructionSets();
-        const target = isets.getInstructionSetTarget(compilationInfo.compiler.instructionSet);
-        if (target != null) args?.push('-mtriple=' + target);
+        let target = isets.getInstructionSetTarget(compilationInfo.compiler.instructionSet);
+        const prependArgs: string[] = [];
 
+        // Check if compiler target is overridden with --target.
         for (const arg of compilationInfo.options) {
-            if (arg.startsWith('-mcpu') || arg.startsWith('-march') || arg.startsWith('-mtriple')) {
-                args?.push(arg);
+            if (arg.startsWith('--target=')) {
+                target = arg.replace(/^--target=/, '');
+            }
+        }
+
+        if (target != null) prependArgs.push('-mtriple=' + target);
+
+        let haveCPU = false;
+        let haveM32 = false;
+        for (const arg of compilationInfo.options) {
+            if (arg.startsWith('-march') && target?.startsWith('x86_64')) {
```

Comment:
Target can start with `i386` which means 32-bit mode. You don't even need to pass `-m32` in this case, just passing `-target i386-pc-linux-gnu` makes clang compile 32-bit code.

Which means, you need to check for `x86_64` or `i386` here.

Reply:
I put something in before seeing your edit. Do you think what I've done is sufficient or do you want to see more, and can you suggest what the conditions should be, and whether it is necessary? I'm still thinking in terms of avoiding invoking the compiler.